src/Client.sol

```diff
@@ -75,8 +75,8 @@ contract Client is Initializable, IClient, MulticallUpgradeable, Ownable2StepUpg
             _ensureMaxDeviationIsNotExceeded(size);
         }
 
-        allowances[msg.sender] -= parsedAmount;
-        emit DatacapAllocated(msg.sender, params.to, params.amount);
+        allowances[msg.sender] -= datacapAmount;
+        emit DatacapAllocated(msg.sender, params.to, datacapAmount);
```

Comment:
`params.to` doesn't make sense here, it will always be verifreg's address.
We need to either:
1. Change this to `DatacapSpent` or `DatacapTransferred` and leave only `msg.sender` and `datacapAmount`, or
2. Emit separate DatacapAllocated event per SP and put SP instead of `params.to`.